### PR TITLE
For #26644 - Apply the wallpaper provided text color to the Fenix logo

### LINKED ISSUE #26644: Home screen logo and private browsing button colors should be adapted to wallpapers.
When using certain wallpapers, the Firefox logo and the PB button are not very visible. 

We should have the ability to adapt colors in order to provide a good contrast ratio.

cc: @Kate-Galetski

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21392)

### PR DESCRIPTION
**Pull request by Mugurell**
The Fenix logo from homescreen was split in two separate images with the one
containing just text now being themable along with the private theme button.
They will be using the wallpaper text color when available.

The values for the text height and the top margin are taken from the Figma
designs - https://www.figma.com/file/d7l1iGabQYqnLcnQbHDGzM/Wallpaper-2.0-%2B-Onboarding-v106?node-id=7771%3A217341

Testing before and after the logo seem visually the same but it's not quite pixel
perfect. It may be because the Figma dimensions needed to be rounded.
Here I have applied a red tint just to show the theming working:

| | release or beta |
| --- | --- |
| current |  |
| updated |  |
| updated clear |  |

| | nightly |
| --- | --- |
| current |  |
| updated |  |
| updated clear |  |

| | debug |
| --- | --- |
| current |  |
| updated |  |
| updated clear |  |

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #26644

### CODE REVIEW

**Review by MatthewTighe — COMMENTED** [review]
Small suggestion, but this otherwise looks good to me. I would like to make sure design can review your concerns before landing though

File: app/src/nightly/res/values/dimens.xml
Comment (MozillaNoah) [inline app/src/nightly/res/values/dimens.xml:8]:
```suggestion
    <dimen name="wordmark_text_height">28dp</dimen>
    <dimen name="wordmark_text_margin_top">12dp</dimen>
```

**Comment by MozillaNoah** [comment]
I wonder if we could save on APK size by replacing the PNGs of the words with vectors. If they turn out to be the same for every version variant except for color, we could just have the one SVG and just re-color it.

**Comment by Kate-Galetski** [comment]
Hi, 

1) It looks to be that the Firefox Preview is not aligned with the Firefox ball. Same goes with Firefox Browser. Nightly looks good.

2) Also could i please see the full homepage screenshot, where logo has the assigned colors: _Light / Text Primary_ or _Light text on color primary._

Screen Shot 2022-09-15 at 11 15 16 AM
Screen Shot 2022-09-15 at 11 15 52 AM

**Comment by Mugurell** [comment]
> It looks to be that the Firefox Preview is not aligned with the Firefox ball. Same goes with Firefox Browser. Nightly looks good.


If testing side by side and drawing straight lines to serve as guidance it seems like the proportions are met.
I also used the same ~[10px top padding](https://github.com/mozilla-mobile/fenix/blob/2d65d56bfbc14b76346fd0007285a31c60ea05eb/app/src/main/res/values/dimens.xml#L82) as seen on [Figma](https://www.figma.com/file/d7l1iGabQYqnLcnQbHDGzM/Wallpaper-2.0-%2B-Onboarding-v106).

(Images can be opened in a new tab to see them bigger)
FirefoxSideBySide

FenixSideBySide

**Comment by Mugurell** [comment]
> Also could i please see the full homepage screenshot, where logo has the assigned colors: Light / Text Primary or Light text on color primary.

Posted videos on Jira - https://mozilla-hub.atlassian.net/browse/FNXV2-21392

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt
Comment (MatthewTighe) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:987]: nit: This shouldn't strictly be necessary because it is checked in `applyWallpaper` as well but I'm fine with keep it if we want the extra assurance

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt
Comment (MatthewTighe) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:991]: Interesting that `ColoStateList` requires an `Int` but other hex values expect `Long`s. I'm not sure there's a good way for us to avoid any errors in the conversion outside of relying on QA

File: app/src/main/java/org/mozilla/fenix/wallpapers/WallpapersUseCases.kt
Comment (MatthewTighe) [inline app/src/main/java/org/mozilla/fenix/wallpapers/WallpapersUseCases.kt:488]: It would be good to verify this storage in a test IMO

**Review by MatthewTighe — APPROVED** [review]
I see @Kate-Galetski left an affirmative comment in Jira, so I'm going to assume this is good from a UX perspective. I would prefer if my comment about a test was addressed, but this otherwise LGTM so r+ in case you want to land when you're online next

**Review by MatthewTighe — COMMENTED** [review]
Ah, it looks like the private browsing color might be used in dark mode which is pretty illegible IMO

dark-mode-text

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt
Comment (Mugurell) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:987]: Will remove it.

**Comment by Mugurell** [comment]
Good catch with the logo color in dark theme, updated now to apply the same default tint as for the private theme when in dark mode:

https://user-images.githubusercontent.com/11428869/191248059-51854007-1d28-426d-bc65-4fe429dbb345.mov

File: app/src/main/java/org/mozilla/fenix/wallpapers/WallpapersUseCases.kt
Comment (Mugurell) [reply app/src/main/java/org/mozilla/fenix/wallpapers/WallpapersUseCases.kt]: Added tests.

**Comment by Mugurell** [comment]
Some unit test are failing with seemingly no relation to these changes 🤔
https://firefoxci.taskcluster-artifacts.net/DzvUnQygTMWIknkdULRZrw/0/public/reports/test/testDebugUnitTest/index.html

**Comment by Mugurell** [comment]
> Some unit test are failing with seemingly no relation to these changes 🤔 https://firefoxci.taskcluster-artifacts.net/DzvUnQygTMWIknkdULRZrw/0/public/reports/test/testDebugUnitTest/index.html

Failures seemed to be linked to the new tests I added in `HomeFragmentTest` - https://github.com/mozilla-mobile/fenix/blob/fc687e89a44c399f63ab5bfd4a27e9bd44c866c6/app/src/test/java/org/mozilla/fenix/home/HomeFragmentTest.kt.
Locally, running all the tests in that class alone works, running all tests with `./gradlew testDebugUnitTest` does indeed show the same failing tests as on TC. Not sure why they fail. To unblock merging I dropped these changes.

**Comment by MatthewTighe** [comment]
@mergify backport releases_v106.0.0

**Comment by MatthewTighe** [comment]
QA has verified so I created and merged a backport

**Comment by Mugurell** [comment]
> QA has verified so I created and merged a backport

Think we also need the followup - https://github.com/mozilla-mobile/fenix/pull/27105, I'll create a backport PR for it and let for you to decide.

**Pull request merged**
